### PR TITLE
added package: adwaita-dark-amoled gtk theme

### PR DESCRIPTION
**Pull request by quantenzitrone**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Ehm what? I have tested this on my NixOS machine, and it is just a GTK theme, so what should differ?

### CODE REVIEW

File: pkgs/data/themes/adwaita-dark-amoled/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  '';
+
+  preferLocalBuild = true;
+
+  meta = with lib; {
+    description = "A full black GTK theme, based on Adwaita.";
+    homepage = "https://gitlab.com/tearch-linux/artworks/themes-and-icons/adwaita-dark-amoled";
+    license = licenses.gpl3;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.quantenzitrone ];
```
Comment (dit7ya) [inline pkgs/data/themes/adwaita-dark-amoled/default.nix:27]:
```suggestion
    maintainers = with maintainers; [ quantenzitrone ];
```

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -10641,6 +10641,12 @@
     githubId = 35086;
     name = "Jonathan Wright";
   };
+  quantenzitrone = {
+    email = "quantenzitrone@protonmail.com";
+    github = "Quantenzitrone";
```
Comment (dit7ya) [inline maintainers/maintainer-list.nix:10646]: Please also add githubId.

**Review by dit7ya — CHANGES_REQUESTED** [review]
Please follow the contribution guidelines. Use a separate commit for adding yourself as maintainer.

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (Quantenzitrone) [reply maintainers/maintainer-list.nix:10646]: But it is stated as optional in the comment at the beginning of maintainers/maintainers-list.nix ... ?

**Comment by Quantenzitrone** [comment]
> Please follow the contribution guidelines. Use a separate commit for adding yourself as maintainer.
> 
> https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md

I can't find anything about adding myself as a maintainer needing to be a separate commit 🤔

File: pkgs/data/themes/adwaita-dark-amoled/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  installPhase = ''
+    mkdir -p $out/share/themes/Adwaita-dark-amoled
+    cp -r . $out/share/themes/Adwaita-dark-amoled
+  '';
+
+  preferLocalBuild = true;
+
+  meta = with lib; {
+    description = "A full black GTK theme, based on Adwaita.";
```
Comment (dit7ya) [inline pkgs/data/themes/adwaita-dark-amoled/default.nix:23]: meta.description should not end with a period

**Review by dit7ya — COMMENTED** [review]
https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers

**Comment by Quantenzitrone** [comment]
So what do I do about it now? Do I have to uncommit my changes and commit them separately with the commit messages also like in the manual?

**Comment by Quantenzitrone** [comment]
I tried this out in a test repository and in theory I should easily be able to reset and recommit my changes here too.

**Comment by dit7ya** [comment]
force push to this PR's branch

**Comment by Quantenzitrone** [comment]
like that?

**Comment by dit7ya** [comment]
Yes, the other commit message should be `adwaita-dark-amoled-theme: init at 2022-07-11`

File: pkgs/data/themes/adwaita-dark-amoled/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    mkdir -p $out/share/themes/Adwaita-dark-amoled
+    cp -r . $out/share/themes/Adwaita-dark-amoled
+  '';
+
+  preferLocalBuild = true;
+
+  meta = with lib; {
+    description = "A full black GTK3 theme, based on Adwaita";
+    homepage = "https://gitlab.com/tearch-linux/artworks/themes-and-icons/adwaita-dark-amoled";
+    license = licenses.gpl3;
```
Comment (romildo) [inline pkgs/data/themes/adwaita-dark-amoled/default.nix:25]: You should be more specific here: is the license `gpl3Only` or `gpl3Plus`?

**Review by romildo — CHANGES_REQUESTED** [review]

File: pkgs/data/themes/adwaita-dark-amoled/default.nix (hunk 4 above)
Comment (Quantenzitrone) [reply pkgs/data/themes/adwaita-dark-amoled/default.nix:25]: As far as I can tell, there is nothing specified in the repository. What is the default fallback, only or plus?

Comment (Quantenzitrone) [reply pkgs/data/themes/adwaita-dark-amoled/default.nix:25]: Oh wait, it is. It's just hidden in between the GPL text.

**Comment by Quantenzitrone** [comment]
can this be merged soon?

**Pull request closed without merging**